### PR TITLE
Get the agent tests running in CI

### PR DESCRIPTION
**Pull request by apotterri**

### CODE REVIEW

File: src/main/java/com/appland/appmap/transform/annotations/Hook.java [hunk 1]
```diff
@@ -183,8 +183,9 @@ public static void apply(CtBehavior targetBehavior, List<HookSite> hookSites) {
             ClassPool.getDefault().get("com.appland.appmap.process.ExitEarly"));
       }
 
+      String exc_invocation = invocations[MethodEvent.METHOD_EXCEPTION.getIndex()];
       targetBehavior.addCatch("{"
-          + invocations[MethodEvent.METHOD_EXCEPTION.getIndex()]
+          + (exc_invocation != null ? exc_invocation : "")
```
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply src/main/java/com/appland/appmap/transform/annotations/Hook.java]: This is good. At one point the strings in the `invocations` array were guaranteed not to be `null`. I've added a commit to make sure this doesn't leak into the before/after inserts as well.

File: src/main/java/com/appland/appmap/transform/annotations/Hook.java [hunk 2]
```diff
@@ -204,6 +199,45 @@ public static void apply(CtBehavior targetBehavior, List<HookSite> hookSites) {
     }
   }
 
+  /* Concatenates potentially null strings with no delimeter. The return value
+     is guaranteed to be non-null.
+  */
+  private static String safeConcatStrings(String... strs) {
```
Comment (apotterri) [inline src/main/java/com/appland/appmap/transform/annotations/Hook.java:205]: Maybe I've just spent too much time writing Ruby, but this looks gross to me.

How would you feel about:
```
    return Arrays.stream(strs)
      .filter(Objects::nonNull)
      .reduce("", (memo, s) -> memo + s.toString());
  }
```

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[submodule "test/test_helper/bats-assert"]
+	path = test/test_helper/bats-assert
+	url = https://github.com/ztombol/bats-assert.git
```
Comment (apotterri) [inline .gitmodules:3]: This isn't the right repo. It should be https://github.com/bats-core/bats-assert .

File: test/smoke.bats [hunk 4]
```diff
@@ -27,12 +33,12 @@ assert_json_contains() {
   # discussion here: https://github.com/stedolan/jq/issues/24)
   local query="${1?} | select(. == null | not)"
 
-  [[ -v DEBUG_JSON ]] && jq <<< "${output}" >&3
+  [[ ! -z "${DEBUG_JSON}" ]] && jq <<< "${output}" >&3
```
Comment (apotterri) [inline test/smoke.bats:36]: Ugh, sorry about this. I forgot that I installed a newer version of Bash using Homebrew....

**Pull request merged**